### ISSUE TITLE
Reconsider payload telemetry document IDs

### ISSUE THREAD

**Issue by adamgreig** [ukhas/habitat#159]
Accorind to http://wiki.apache.org/couchdb/Performance we are basically doing almost the worst possible thing (non-sequential, entirely random, long) for our most common (payload telemetry) type of document.

However, hashing still has a whole ton of advantages -- it's the only way we can be sure to avoid multiple documents for the same received data and works very well with the MVCC system in Couch. Using Couch IDs would require racey database lookups to see if someone has already made the document you want to save; using something generated server-side has similar issues if we want it to be monotonic. Either way we lose the ability to generate the ID given the data.

One easy change would be to use base64 encoded IDs instead of hexadecimal IDs: on sha256, this moves us from 64 bytes to 44 bytes (a ~31% reduction). We could also move to MD5 instead of SHA256, giving 24 byte IDs (a 63% reduction) that would probably still be sufficiently unique (approximately 5E21 documents until 50% chance of collision -- millions of years of telemetry).

However, the changeover would be somewhat traumatic. We'd have to rewrite the current transition stuff, the new dl-fldigi stuff, anything else that relies on the assumption that the docs will be sha256 sums, and then move over all the telem in the database. Not impossible, but not a thing to be done trivially, either.

**Comment by adamgreig** [comment]
After some extensive testing and discussion, it turns out that:
- Using an MD5 in base64 instead of SHA256 in hex from the get-go would have made more sense as disk usage is smaller
- Using parts of the actual data in the hope that it's sequential also helps with disk usage
- But overall, we're talking a saving of 50% of the disk usage, which while significant, is not a limiting factor for us right now
- The effort taken transitioning could be better spent elsewhere
- If this does become a bottleneck, this change alone is unlikely to be sufficient to fix it
- Finally, we can make up almost all the disk space by regenerating the database with sorted keys, a slow operation but one that gives us the minimum space consumption

In conclusion, we're sticking with SHA256 hex digested checksums for message IDs, because even though MD5 in base64 would have been a slightly better starting choice, the effort to transition isn't worth it.
